src/trio/_tools/gen_exports.py

```diff
@@ -302,7 +302,7 @@ def process(files: Iterable[File], *, do_test: bool) -> None:
             print("Generated sources are up to date.")
     else:
         for new_path, new_source in new_files.items():
-            Path(new_path).write_text(new_source, encoding="utf-8", newline="\n")
+            Path(new_path).write_text(new_source, encoding="utf-8")
```

Comment:
At least with `open`, `newline` is a good idea for Windows users so that `gen_exports.py` won't output files using `"\r\n"`. I'm not sure if that's the case or not with `Path#write_text`.

Reply:
Apparently, Path.write_text's `newline` argument was only added in 3.10 or something

Reply:
Probably worth switching back to `open` then and then making an issue on ruff about availability of the parameter.

Reply:
Fixed in 924b6dd